js/keyboard/keyboard.js

```diff
@@ -0,0 +1,147 @@
+import {
+	LANGUAGE,
+} from '../constants.js';
+import buttons from './buttonData.js';
+import Button from './button.js';
+
+export default class Keyboard {
+	constructor(outputEl, language) {
+		this.language = language || LANGUAGE.EN;
+		this.outputEl = outputEl;
+		this.buttons = [];
+		this.isShifted = false;
+		this.el = document.createElement('div');
+	}
+
+	draw() {
+		this.el.classList.add('keyboard');
+		buttons.forEach((row) => {
+			const rowEl = document.createElement('div');
+			this.el.append(rowEl);
+			rowEl.classList.add('row');
+			row.forEach((buttonOpts) => {
+				const button = new Button({
+					language: this.language,
+					...buttonOpts,
+				});
+				this.buttons.push(button);
+				rowEl.append(button.draw());
+			});
+		});
+
+		return this.el;
+	}
+
+	attachEvents() {
+		const addCharacter = (charValue) => {
+			const {
+				value,
+				selectionStart,
+				selectionEnd,
+			} = this.outputEl;
+			this.outputEl.focus();
+			if (value.length === selectionEnd) {
+				this.outputEl.value += charValue;
+			} else {
+				this.outputEl.value = value.substring(0, selectionStart)
+					+ charValue
+					+ value.substring(selectionEnd);
+				this.outputEl.setSelectionRange(selectionEnd + 1, selectionEnd + 1);
+			}
+		};
+
+		const removeCharacter = (position) => {
+			const {
+				value,
+				selectionStart,
+				selectionEnd,
+			} = this.outputEl;
+			this.outputEl.focus();
+			if (selectionStart === selectionEnd) {
+				this.outputEl.value = value.substring(0, position) + value.substring(position + 1);
+				this.outputEl.setSelectionRange(position, position);
+			} else {
+				this.outputEl.value = value.substring(0, selectionStart) + value.substring(selectionEnd);
+				this.outputEl.setSelectionRange(selectionStart, selectionStart);
+			}
+		};
+
+		const toggleKeys = (shiftKey) => this.el.classList.toggle('keyboard_isShifted', shiftKey ^ this.isCapsLock);
+
+		document.addEventListener('keydown', (e) => {
+			const pushedButton = this.buttons.find((x) => x.code === e.code);
+			if (pushedButton) {
+				pushedButton.el.classList.add('button_active');
+			}
```

Comment:
It would be better add method (for example) `activateButton` to the class `Button` and call it here like
``` javascript
activateButton(e.code);
```
All the buttons' info will be used only in the class `Button`